skbuild/cmaker.py

```diff
@@ -87,6 +88,32 @@ def get_cmake_version(cmake_executable=CMAKE_DEFAULT_EXECUTABLE):
     return version_string.splitlines()[0].split(' ')[-1]
 
 
+class push_env_without_destdir(push_env):
+    """Remove DESTDIR from MAKEFLAGS and environment
+
+    When scikit-build is invoked as a part of a larger cmake project, makefile,
+    or something else that sets the DESTDIR environment variable, the internal
+    cmake --target install honours it and makes artefacts under DESTDIR/
+
+    The internal install into _skbuild/ should never need to respect DESTDIR,
+    so simply remove it from the environment. When cmake generates the
+    makefiles, it stores the make parameters [1] in the MAKEFLAGS environment
+    variable.  The DESTDIR entry needs to be removed from here as well.
+
+    [1] make DESTDIR=/path install
+    """
```

Comment:
What do you think of removing the constructor argument and doing the following:

```python
    def __init__(self):
        makeflags = os.environ.get('MAKEFLAGS', None)

        if makeflags is not None:
            flags = [flag for flag in makeflags.split(' ') if not flag.startswith('DESTDIR=')]
            makeflags = ' '.join(flags)

        super(push_env_without_destdir, self).__init__(DESTDIR=None, MAKEFLAGS=makeflags)
```

Reply:
It still wouldn't all callers of cmaker.make() to pass a specific env. I'm not aware of this feature being _used_ at all (if it is not, why is the env argument bubbled out?). If the env argument to make() can go away, so do all of these problems and the context manager will become a lot simpler.

Reply:
Good point ... I missed that. Sorry for the noise.

The env parameter is indeed important and your implementation makes sense.

https://github.com/scikit-build/scikit-build/blob/0c33f15b0c0452e8bd70ab6057b4d283d6025b51/skbuild/setuptools_wrap.py#L579-L589